src/app/core/submission/vocabularies/vocabulary-entries-response-parsing.service.spec.ts

```diff
@@ -0,0 +1,111 @@
+import { getMockObjectCacheService } from '../../../shared/mocks/object-cache.service.mock';
+import { ErrorResponse, GenericSuccessResponse } from '../../cache/response.models';
+import { DSpaceRESTV2Response } from '../../dspace-rest-v2/dspace-rest-v2-response.model';
+import { VocabularyEntriesResponseParsingService } from './vocabulary-entries-response-parsing.service';
+import { VocabularyEntriesRequest } from '../../data/request.models';
+
+describe('VocabularyEntriesResponseParsingService', () => {
+  let service: VocabularyEntriesResponseParsingService;
+  const metadata = 'dc.type';
+  const collectionUUID = '8b39g7ya-5a4b-438b-851f-be1d5b4a1c5a';
+  const entriesRequestURL = `https://rest.api/rest/api/submission/vocabularies/types/entries?metadata=${metadata}&collection=${collectionUUID}`
+
+  beforeEach(() => {
+    service = new VocabularyEntriesResponseParsingService(getMockObjectCacheService());
+  });
+
+  describe('parse', () => {
+    const request = new VocabularyEntriesRequest('client/f5b4ccb8-fbb0-4548-b558-f234d9fdfad6', entriesRequestURL);
+
+    const validResponse = {
+      payload: {
+        _embedded: {
+          entries: [
+            {
+              display: 'testValue1',
+              value: 'testValue1',
+              otherInformation: {},
+              type: 'vocabularyEntry'
+            },
+            {
+              display: 'testValue2',
+              value: 'testValue2',
+              otherInformation: {},
+              type: 'vocabularyEntry'
+            },
+            {
+              display: 'testValue3',
+              value: 'testValue3',
+              otherInformation: {},
+              type: 'vocabularyEntry'
+            },
+            {
+              authority: 'authorityId1',
+              display: 'testValue1',
+              value: 'testValue1',
+              otherInformation: {
+                id: 'VR131402',
+                parent: 'Research Subject Categories::SOCIAL SCIENCES::Social sciences::Social work',
+                hasChildren: 'false',
+                note: 'Familjeforskning'
+              },
+              type: 'vocabularyEntry',
+              _links: {
+                vocabularyEntryDetail: {
+                  href: 'https://rest.api/rest/api/submission/vocabularyEntryDetails/srsc:VR131402'
+                }
+              }
+            }
+          ]
+        },
+        _links: {
+          first: {
+            href: 'https://rest.api/discover/browses/author/entries?page=0&size=5'
```

Comment:
The pagination in this example response is incorrect/invalid.  This is the `/submission/vocabularies/types/entries` endpoint, so it should never respond with pagination for `/discover/browses/author/entries`.  So, we need to fix this response to be a valid response that we'd see from the REST API.

Reply:
they should be fixed

Reply:
@atarix83 : It looks like you fixed the wrong part of the links.  All the links in that file still refer to `/discover/browses/author/entries` endpoint.....but the tests are testing an entirely different endpoint.  The tests are testing `/submission/vocabularies/types/entries`.  So, these still require updating.

Reply:
done